examples/rustc-driver-interacting-with-the-ast.rs

```diff
@@ -76,7 +77,7 @@ fn main() {
                                     let hir_id = expr.hir_id; // hir_id identifies the string "Hello, world!"
                                     let def_id = tcx.hir().local_def_id(item.hir_id()); // def_id identifies the main function
                                     let ty = tcx.typeck(def_id).node_type(hir_id);
-                                    println!("{:?}: {:?}", expr, ty); // prints expr(HirId { owner: DefIndex(3), local_id: 4 }: "Hello, world!"): &'static str
+                                    println!("{:?}: {:?}", expr, ty); // prints "`rustc_hir::Expr`: `rustc_middle::ty::Ty`"
```

Comment:
are you sure? that seems very strange for it to print, normally it gives details about the item itself

Reply:
It actually prints:
```
fn main() { let message = "Hello, world!"; println!("{}", message); }
Expr { hir_id: HirId { owner: DefId(0:3 ~ rust_out[a9df]::main), local_id: 2 }, kind: Lit(Spanned { node: Str("Hello, world!", Cooked), span: <main.rs>:1:27: 1:42 (#0) }), span: <main.rs>:1:27: 1:42 (#0) }: &str
```

Well, I remember I truncated the output because it's a bit long but... yeah it's quite different.
Any suggestions here? I feel it's verbose and would like to shorten if possible. Though I don't mind pasting it as-is if you think it's better :)

Reply:
I would rid of the comment

Reply:
Removed!